src/t_stream.c

```diff
@@ -2775,6 +2775,11 @@ void xsetidCommand(client *c) {
     if (o == NULL || checkType(c,o,OBJ_STREAM)) return;
     stream *s = o->ptr;
 
+    if (streamCompareID(&id,&s->max_deleted_entry_id) < 0) {
+        addReplyError(c,"The ID specified in XSETID is smaller than current max_deleted_entry_id");
```

Comment:
please add a test

Reply:
> please add a test

Test case is added, please check it, Thanks